src/box/memtx_tx.c

```diff
@@ -1910,10 +1918,28 @@ memtx_tx_tuple_clarify_impl(struct txn *txn, struct space *space,
 		if (memtx_tx_story_is_visible(story, txn, is_prepared_ok,
 					      &result, &own_change))
 			break;
+		if (story->add_psn != 0 && story->add_stmt != NULL &&
+		    txn != NULL) {
+			/*
+			 * If we skip prepared story then the transaction
+			 * must be before prepared in serialization order.
+			 * That can be a read view or conflict already.
+			 */
+			memtx_tx_handle_conflict(story->add_stmt->txn, txn);
```

Comment:
Please consider updating the documentation of `memtx_tx_handle_conflict`:
https://github.com/tarantool/tarantool/blob/f1d750f036956873c79bf5de51f77b38da25ae0f/src/box/memtx_tx.h#L215
Here we are handling a conflict after preparation of transaction.

Reply:
Rewrote that comment.